examples/avro_generic_consumer_example/avro_generic_consumer_example.go

```diff
@@ -0,0 +1,128 @@
+// Example function-based high-level Apache Kafka consumer
+package main
+
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// consumer_example implements a consumer using the non-channel Poll() API
+// to retrieve messages and events.
+
+import (
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/confluentinc/confluent-kafka-go/schemaregistry"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func main() {
+
+	if len(os.Args) < 5 {
+		fmt.Fprintf(os.Stderr, "Usage: %s <schema-registry> <broker> <group> <topics..>\n",
+			os.Args[0])
+		os.Exit(1)
+	}
+
+	url := os.Args[1]
+	broker := os.Args[2]
+	group := os.Args[3]
+	topics := os.Args[4:]
+	sigchan := make(chan os.Signal, 1)
+	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)
+
+	c, err := kafka.NewConsumer(&kafka.ConfigMap{
+		"bootstrap.servers": broker,
+		// Avoid connecting to IPv6 brokers:
+		// This is needed for the ErrAllBrokersDown show-case below
+		// when using localhost brokers on OSX, since the OSX resolver
+		// will return the IPv6 addresses first.
+		// You typically don't need to specify this configuration property.
+		"broker.address.family": "v4",
+		"group.id":              group,
+		"session.timeout.ms":    6000,
+		"auto.offset.reset":     "earliest"})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to create consumer: %s\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Created Consumer %v\n", c)
+
+	deser := schemaregistry.GenericAvroDeserializer{}
+	err = deser.Configure(&kafka.ConfigMap{
```

Comment:
First up, I don't have a clear view on how configuration should be done in a broader sense for this client. But here are some things that come to mind on that atm:

- `ConfigMap` has some very librdkafka specific stuff in it - it doesn't seem right to me that it is reused to configure the serdes.
- I don't reckon I even think this is a good design choice for the clients - i think a plain old map (or something else idiomatic that isn't librdkafka specific) would probably have been a better - encapsulate the librdkafka stuff further down. And a manifestation of this being not the best way to do it is that it complicates working out what's best now we have something else to configure. 
- I also don't like how a reference to a reference is passed, superfluous. But do we go for consistency or improvement?

Reply:
Also, i can't quite recall, but won't librdkafka/the go client complain if you give it a config it doesn't know about? which means you'd never create a single config and pass it to both.

Anyway, I think it would pay to think through how best to do config some more. Get a go fanatic to weigh in.